### DISCUSSION TITLE
What is about support to build a component library?

### DISCUSSION THREAD

**Discussion by ClausKlein** [TheLartians/Greeter#101] [Ideas]
see https://github.com/ClausKlein/cmake-example-component-lib/blob/master/README.md

**Comment by ClausKlein** [comment]
@TheLartians what do you think about my template?

**Reply by TheLartians** [reply]
Hey, I think it's a great idea to have another template that shows how to do components! Imo component libraries are a bit niche though (I personally prefer just splitting a library into smaller libraries), but having separate starters for different needs makes complete sense. Perhaps we could add a section "Similar Projects" to the readme showing different starters or forks.

**Reply by ClausKlein** [reply]
But it was necessary to extend the [PackageProject.cmake ](https://github.com/ClausKlein/PackageProject.cmake)module.
